### PR TITLE
Label name check for 'count_values'

### PR DESCRIPTION
**Pull request by codesome**
Fixes https://github.com/prometheus/prometheus/issues/4584

Signed-off-by: Ganesh Vernekar <cs15btech11018@iith.ac.in>

### CODE REVIEW

**Comment by codesome** [comment]
@brian-brazil I added a unit test. Would this be fine?

**Comment by brian-brazil** [comment]
That's fine thanks.

File: promql/engine_test.go [hunk 1]
```diff
@@ -340,6 +345,12 @@ load 10s
 			t.Fatalf("unexpected error creating query: %q", err)
 		}
 		res := qry.Exec(test.Context())
+		if c.ShouldError {
+			if res.Err == nil {
```
[the remark this answers is not in the document]
Comment (codesome) [reply promql/engine_test.go]: Done

**Comment by gouthamve** [comment]
👍

File: promql/engine.go [hunk 2]
```diff
@@ -1494,6 +1495,9 @@ func (ev *evaluator) aggregation(op ItemType, grouping []string, without bool, p
 	var valueLabel string
 	if op == itemCountValues {
 		valueLabel = param.(string)
+		if !model.LabelName(valueLabel).IsValid() {
+			ev.errorf("Invalid Label Name '%s'", valueLabel)
```
Comment (gouthamve) [inline promql/engine.go]: Taking another look, we use `%q` in the other parts of the code-base for quoting strings. Also, I think it should be all small letters, no capitals. Thoughts @brian-brazil?

Comment (brian-brazil) [reply promql/engine.go]: @juliusv was looking at that recently. They should probably all be lower case.

Comment (codesome) [reply promql/engine.go]: @gouthamve I made the changes.

File: promql/engine.go [hunk 3]
```diff
@@ -1494,6 +1495,9 @@ func (ev *evaluator) aggregation(op ItemType, grouping []string, without bool, p
 	var valueLabel string
 	if op == itemCountValues {
 		valueLabel = param.(string)
+		if !model.LabelName(valueLabel).IsValid() {
+			ev.errorf("invalid label name '%q'", valueLabel)
```
Comment (gouthamve) [inline promql/engine.go]: You don't need the `''` anymore.

Comment (codesome) [reply promql/engine.go]: Oh, now I get the use of `%q`. Changed.

File: promql/engine_test.go [hunk 4]
```diff
@@ -340,6 +346,10 @@ load 10s
 			t.Fatalf("unexpected error creating query: %q", err)
 		}
 		res := qry.Exec(test.Context())
+		if c.ShouldError {
+			testutil.NotOk(t, res.Err, "expected error for the query '%s'", c.Query)
```
Comment (gouthamve) [inline promql/engine_test.go]: Same here, sorry :/

Comment (codesome) [reply promql/engine_test.go]: Haha, done!

**Pull request merged**